src/types/io.ts

```diff
@@ -565,7 +565,23 @@ export type DemandFactorSettings = {
 
 // Interfaces
 
-export interface AoARIORead {
+// simple interface to allow multiple implementations of ArNSNameResolver
+export type ArNSNameResolutionData = {
+  name: string;
+  owner?: string; // could be unowned
+  txId: string;
+  type: 'lease' | 'permabuy';
+  processId: string;
+  ttlSeconds: number;
+  recordIndex?: number; // TODO: the SDK should always provide a priority index, even if the ANT does not have a priority set
```

Comment:
should this be non-nullable as the retrun from SDK? the comment seems to suggest so

also why the change from `priority` to `recordIndex`?

Reply:
I added other TODOs to enforce priority is set on all ant `getRecords` requests. I'd like to separate that change and test in isolation. After that I'll come back and remove these `todos`

this API essentially normalizes the arns name data on the ARIO contract and the record data from the ANT. 


> also why the change from priority to recordIndex?

priority is a overloaded, but i agree it should be consistent. will modify